clustermesh-apiserver/coredns-mcsapi-auto-configure/cmd_test.go

```diff
@@ -0,0 +1,206 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package corednsMCSAPIAutoConfigure
+
+import (
+	"log/slog"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestUpdateCorefiles(t *testing.T) {
```

Comment:
You could use a script test to validate the entire cell, instead of only the `updateCorefile` function.